pulpcore/cli/rpm/repository.py

```diff
@@ -224,6 +224,10 @@ def sync(
     if optimize:
         body["optimize"] = optimize
     if skip_types:
+        if "treeinfo" in skip_types:
+            repository_ctx.pulp_ctx.needs_plugin(
+                PluginRequirement("rpm", min="3.19.0", feature="--skip-type treeinfo")
+            )
```

Comment:
I'd prefer to see this check in the context layer (aka, the great library of Pulp version specifics).